src/mem/freelist.h

```diff
@@ -59,29 +59,44 @@ namespace snmalloc
   {
     FreeObject* next_object;
 
-    static FreeObject* encode(uintptr_t local_key, FreeObject* next_object)
+    /**
+     * On architectures which use IntegerPointers, we can obfuscate our free
+     * lists and use this to drive some probabilistic checks for integrity
+     */
+    template<typename AAL = Aal, typename T = FreeObject>
+    static std::enable_if_t<aal_supports<IntegerPointers, AAL>, T*>
```

Comment:
Can you make this
```
static std::enable_if_t<aal_supports<IntegerPointers, AAL> && CHECK_CLIENT, T*>
```
?

Then you can remove the `ifdef` inside, and use the second encode for that case.